File: app/src/main/java/com/fanhl/aidldemo/client/MainAdapter.java
package com.fanhl.aidldemo.client;

import android.view.View;
import android.widget.TextView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.fanhl.aidldemo.Msg;

import java.util.Date;

public class MainAdapter extends BaseQuickAdapter<Msg, MainAdapter.ViewHolder> {
    public MainAdapter() {
        super(android.R.layout.simple_list_item_1);
    }

    @Override
    protected void convert(ViewHolder helper, Msg item) {
        helper.bind(item);
    }

    public static class ViewHolder extends BaseViewHolder {
        public ViewHolder(View view) {
            super(view);
        }

        public void bind(Msg data) {
            ((TextView) itemView.findViewById(android.R.id.text1)).setText(new Date(data.getTime()) + ": " + data.getMsg());
        }
    }
}

File: shared/src/main/java/com/fanhl/aidldemo/shared/MainAdapter.kt
package com.fanhl.aidldemo.shared

import android.view.View
import android.widget.TextView
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import java.util.*

class MainAdapter : BaseQuickAdapter<Msg, MainAdapter.ViewHolder>(android.R.layout.simple_list_item_1) {
    override fun convert(helper: ViewHolder?, item: Msg?) {
        helper?.bind(item)
    }

    class ViewHolder(view: View?) : BaseViewHolder(view) {
        fun bind(data: Msg?) {
            data ?: return
            (itemView.findViewById<View>(android.R.id.text1) as TextView).text = Date(data.time).toString() + ": " + data.msg
        }
    }
}